x/cheqd/utils/did.go

```diff
@@ -6,6 +6,7 @@ import (
 )
 
 var DidForbiddenSymbolsRegexp, _ = regexp.Compile(`^[^#?&/\\]+$`)
+var DidAllowedSymbolsRegexp, _ = regexp.Compile(`^[a-zA-Z0-9_]*$`)
```

Comment:
I would use an actual Base58 check, instead of relying on just checking the character set: https://github.com/anaskhan96/base58check (there might be other libraries). For example, Base58 _doesn't_ allow ambiguous characters, such as `0` vs 'O` etc. https://en.wikipedia.org/wiki/Binary-to-text_encoding#Base58

See Base58 alphabet here: https://en.bitcoinwiki.org/wiki/Base58#Alphabet_Base58

I don't think you need to check for forbidden symbols, just check for allowed symbols.